prac_10/testing.py

```diff
@@ -0,0 +1,63 @@
+"""
+CP1404/CP5632 Practical
+Testing demo using assert and doctest
+"""
+
+import doctest
+from prac_07.car import Car
+
+
+def repeat_string(s, n):
+    """Repeat string s, n times, with spaces in between."""
+    if n <= 1:
+        return s
+    else:
+        return " ".join([s, repeat_string(s, n - 1)])
+
+
+def is_long_word(word, length=5):
+    """
+    Determine if the word is as long or longer than the length passed in
+    >>> is_long_word("not")
+    False
+    >>> is_long_word("supercalifrag")
+    True
+    >>> is_long_word("Python", 6)
+    True
+    """
+    return len(word) >= length
+
+
+def phrase_to_sentence(phrase=""):
+    """
+    Reformat phrase to sentance starting with capital and ending with full stop.
+    >>> phrase_to_sentence("")
+    ''
+    >>> phrase_to_sentence("hi")
+    'Hi.'
+    >>> phrase_to_sentence("hello")
+    'Hello.'
+    >>> phrase_to_sentence("It is an ex parrot.")
+    'It is an ex parrot.'
+    >>> phrase_to_sentence("pRograms r 4 Programmers")
+    'PRograms r 4 Programmers.'
+    >>> phrase_to_sentence("012345")
+    '012345.'
+    """
+    return phrase[0:1].upper() + phrase[1:] + ("." if phrase[-1:] != "." else "") if phrase != "" else ""
```

Comment:
This could be clearer. The first expression should be `phrase[0]`, not a slice of only 1 character. I would separate the last if as a special case.